modules/tensor_mechanics/include/materials/ComputeGeneralStressBase.h

```diff
@@ -0,0 +1,59 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+#include "RotationTensor.h"
+#include "DerivativeMaterialInterface.h"
+
+/**
+ * ComputeGeneralStressBase is the base class for stress tensors,
+ * whether they are computed by MOOSE strain calculators (ComputeStressBase) or external user
+ * plug-ins.
```

Comment:
```suggestion
 * ComputeGeneralStressBase is the base class for stress calculator materials that perform
 * their own strain calculations (like the UMAT plugins) rather than using strain tensors
 * computed by separate MOOSE material objects (those classes should derivefrom ComputeStressBase) .  
```

Reply:
I have reworded that class comment with more details. It's not straightforward to explain since UMAT also employs MOOSE's strain calculators, but it can potentially make use of quantities on the displaced mesh, such as COORD.